lib/9pfs/README.md

```diff
@@ -0,0 +1,166 @@
+* The `readdir_off` refers to the offset from within the `readdir_buf` where we can find information about the next child.
+
+### Note Data Structure
+
+```c
+struct uk_9pfs_node_data {
+ /* File id of the vfs node */
+ struct uk_9pfid        *fid;
+ /* Number of files opened from the vfs node */
+ int                    nb_open_files;
+ /* Is a 9P remove call required when `nb_open_files` reaches 0? */
+ bool                   removed;
+};
+```
+
+The `uk_9pfs_node_data` structure is used for operations that involve allocating or freeing `vnode` data.
+
+## Configuring Applications to Use `9pfs`
+
+An Unikraft application that uses the `9pfs` filesystem will use code snippets such as the one below:
+
+```c
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[])
+{
+   char buf[1024];
+   int fd;
+
+   fd = open("/westworld.txt", O_RDWR | O_CREAT, 0777);
+   write(fd, "These violent delights have violent ends.", 41);
+   close(fd);
+
+   fd = open("/westworld.txt", O_RDWR | O_CREAT, 0777);
+   read(fd, buf, 41);
+   printf("%s\n", buf);
+   close(fd);
+
+   return 0;
+}
+```
+
+To configure the application to run under `9pfs` we follow the steps below:
+
+1. Enter the configuration interface by running:
+
+   ```console
+   $ make menuconfig
+   ```
+
+1. Select `vfscore:` `VFS Core Interface`.
+1. Under `vfscore: Configuration  --->` select `Automatically mount a root filesystem (/)`.
+1. Select `Default root filesystem -> 9pfs`.
+1. For the `Default root device` option fill the `fs0` string (instead of the default `rootfs` string).
+1. We want to run Unikraft with QEMU / KVM, so we must select KVM guest in the `Platform Configuration` menu. For `9pfs` we also need to enable, in the KVM guest options menu, `Virtio --->`, then `Virtio PCI device support`.
+1. Save the configuration and exit the interface.
+1. Create a new folder in the current directory by running:
+
+   ```console
+   $ mkdir fs0
+   ```
+
+The complete command for running this program with `qemu` is:
+
+```console
+$ qemu-system-x86_64 -fsdev local,id=myid,path=./fs0/,security_model=none -device virtio-9p-pci,fsdev=myid,mount_tag=fs0 -kernel build/06-adding-filesystems_kvm-x86_64 -nographic
+```
+
+Breaking it down we can see:
+
+* `-fsdev local,id=myid,path=./fs0/,security_model=none` - assign an id (`myid`) to the `./fs0/` local folder.
+* `-device virtio-9p-pci,fsdev=myid,mount_tag=fs0` - create a device with the 9pfs type, assign the `myid` for the `-fsdev` option and also assign the mount tag that we configured above (`fs0`).
+  Unikraft will look after that mount tag when trying to mount the filesystem, so it is important that the mount tag from the configuration is the same as the one given as argument to qemu.
+* `-kernel build/app-helloworld_kvm-x86_64` - tells QEMU that it will run a kernel; if this parameter is omitted, QEMU will think it runs a raw file.
```

Comment:
```suggestion
* `-kernel build/app-helloworld_kvm-x86_64` - tells QEMU that it will run a kernel image; if this parameter is omitted, QEMU will think it runs a raw file.
```